Develop1

Cammelen opened the pull request:
1. Добавлены библиотеки:

- Junit 4
- Mockito
- Jacoco

2. Написаны тесты на классы:

- Bun (BunTest)
- Burger (BurgetTest)
- Ingredient (IngredientTest)
- IngredientType (IngredientTypeTest)

3. Добавлен базовый тестовый класс TestBase
4. Добавлен отчет Jacoco

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Diplom_1 commit

Commit:
Последний коммит, в который добавлен файл README.md

nikolay-977 commented on pom.xml:
```diff
@@ -9,8 +9,66 @@
     <properties>
-        <maven.compiler.source>8</maven.compiler.source>
-        <maven.compiler.target>8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
```
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (mvn clean test). Как оформлять README.md можно ознакомиться в статье https://dzen.ru/a/YbzudflBPE1IDJF-

Cammelen replied:
Файл README.md был добавлен в "Последний коммит, в который добавлен файл README.md"
https://github.com/Cammelen/Diplom_1/pull/1/commits/7640250c59db4c79f0c3c65069c4b514e39ac1fd

nikolay-977 commented on pom.xml:
```diff
@@ -9,8 +9,66 @@
     <version>1.0-SNAPSHOT</version>
 
     <properties>
```
Можно улучшить: в заготовке проекта есть лишние файлы, такие как test/java/tests/read.me. Их следует удалять.

Cammelen replied:
Исправлено в коммите "Second_commit"
https://github.com/Cammelen/Diplom_2/pull/1/commits/328f3783ef8244df916454f4afd8995e7dba1f9a

nikolay-977 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,73 @@
+        this.price = price;
+    }
+
+    @Parameterized.Parameters
```
Можно улучшить: В параметризованных тестах для аннотации `@Parameterized.Parameters` лучше использовать аргумент name: `@Parameterized.Parameters(name = "Стоимость булочки. Тестовые данные: {0} {1}")`, где {0}, {1} - индексы параметров. С аргументом name наименование теста отобразится в отчете. А использование индексов параметров повысит информативность проверки.

Cammelen replied:
Исправлено в коммите "Second_commit"
https://github.com/Cammelen/Diplom_2/pull/1/commits/328f3783ef8244df916454f4afd8995e7dba1f9a

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,141 @@
+    @Mock
+    Bun bun;
+
+    Ingredient ingredientSauce = new Ingredient(IngredientType.SAUCE, "hot sauce", 100);
```
Необходимо исправить: Тесты для класса Burger должны быть изолированы от классов Bun и Ingredient. При написании тестов для класса Burger во всех кейсах используй моки (тема: Юнит-тесты > методы mockito) и стабы (тема: Юнит-тесты > что такое стаб) классов Bun и Ingredient, а не реальные объекты.

Cammelen replied:
Исправлено в коммите "Second_commit"
https://github.com/Cammelen/Diplom_2/pull/1/commits/328f3783ef8244df916454f4afd8995e7dba1f9a

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,141 @@
+        burger.removeIngredient(0);
+
+        assertEquals(ingredientList, burger.ingredients);
+        System.out.println(burger.ingredients);
```
Можно улучшить: если System.out.println() используется для отладки, лучше использовать логирование. https://www.baeldung.com/java-logging-intro

Cammelen replied:
Исправлено в коммите "Second_commit"
https://github.com/Cammelen/Diplom_2/pull/1/commits/328f3783ef8244df916454f4afd8995e7dba1f9a

nikolay-977 commented on src/test/java/IngredientTest.java:
```diff
@@ -0,0 +1,64 @@
+@RunWith(Parameterized.class)
+public class IngredientTest extends TestBase {
+
+    IngredientType ingredientType;
```
Можно улучшить: Для всех полей следует указывать модификатор доступа.

Cammelen replied:
Исправлено в коммите "Second_commit"
https://github.com/Cammelen/Diplom_2/pull/1/commits/328f3783ef8244df916454f4afd8995e7dba1f9a

Commit:
Second_commit

1. Убран лишний файл read.me
2. В паретризованных тестах добавлен аргумент name
3. Тестовый класс Burger изолирован от класса Ingredient
4. В тестовых классах убран метод System.out.println()
5. Для всех полей указан модификатор доступа
